src/lib/classes/lsp8-identifiable-digital-asset.ts

```diff
@@ -22,6 +27,12 @@ import {
   setMetadataAndTransferOwnership$,
 } from '../services/digital-asset.service';
 
+type ObservableOrPromise<
+  T extends ContractDeploymentOptionsReactive | ContractDeploymentOptionsNonReactive
+> = T extends ContractDeploymentOptionsReactive
+  ? Observable<DeploymentEventContract | DeploymentEventTransaction>
+  : Promise<DeployedLSP8IdentifiableDigitalAsset>;
+
```

Comment:
Looks like this part of code is duplicated 3 times in the lspX files. Maybe at least you can give a slightly more explicit name? Like `LSP8ObservableOrPromise`, `LSP7ObservableOrPromise`, etc.

Or create a global type you can reuse?

Reply:
Yeah its tricky to make a reuable type here because the contract of the promise is different (`DeployedLSP8IdentifiableDigitalAsset` etc). Have improved naming

Reply:
Yeah its tricky to make a reuable type here because the contract of the promise is different (`DeployedLSP8IdentifiableDigitalAsset` etc). Have improved naming